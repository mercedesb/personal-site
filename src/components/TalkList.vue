<template>
  <div v-if="talks.length" class="TalkList">
    <smart-link :class="'TalkListItem TalkListItem--' + talk.color + ' ' + talk.iconCssClass" v-for="talk in talks" v-bind="talk" :key="talk.id" :to="'/speaking/' + talk.urlSegment">
      <div class="TalkListItem-icon" v-html="talk.iconSvg"></div>
      <h3 class="TalkListItem-title">{{talk.title}}</h3>
      <div class="TalkListItem-moreInfo">
        Resources
      </div>
    </smart-link>
  </div>
</template>

<script>
export default {
  props: {
    talks: Array 
  }
}
</script>

<style lang="scss">
@import '../assets/styles/variables.scss';

.TalkList {
  display: flex;
  flex-wrap: wrap;
  max-width: 1000px;
  margin: 0 auto;
}

.TalkListItem {
  width: 100%;
  margin: $base-spacing $small-spacing;

  @include media($min-tablet) {
    width: 45%;
  }

  @include media($min-desktop) {
    width: 30%;
  }

  > div, > a {
    display: flex;
    justify-content: space-between;
    align-items: center;
    flex-direction: column;
  }

  &-icon {
    width: 30%;
    margin: $base-spacing 0 $small-spacing;

    @include light-svg;

  }

  &-title {
    font-size: $large-font-size;
    text-align: center;
    color: $white;
    margin: $small-spacing;
  }

  &-moreInfo {
    opacity: 0;
    transition: opacity .5s;
    font-weight: $heavy-font-weight;
    text-decoration: underline;
    color: $white;
    margin: $small-spacing 0 $base-spacing;
  }

  &:hover {
    cursor: pointer;

    .TalkListItem-moreInfo {
      opacity: 1;
      transition: opacity .5s;
    }
  }

  @include background-color;
  @include hover-saturate;

}

</style>
