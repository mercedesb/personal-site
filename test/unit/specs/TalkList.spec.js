import TalkList from '@/components/TalkList.vue'
import { TestUtility } from './TestUtility'

describe('TalkList', () => {
  let component

  const store = {
    state: {
      talks: TestUtility.talks
    }
  }

  const shallow = propsData => TestUtility.shallow(TalkList, {
    store,
    propsData: {
      ...propsData
    }
  })

  describe('Snapshots', () => {
    it('matches snapshot', () => {
      component = shallow()
      expect(component.element).toMatchSnapshot()
    })
  })

  describe('Computed', () => {
    describe('talks', () => {
      beforeEach(() => {
        component = shallow()
      })

      it('has talks populated', () => {
        expect(component.vm.talks).toBeDefined()
      })

      it('sorts the talks in desc order by the most recent talk given', () => {
        expect(component.vm.talks).toEqual([TestUtility.talks[1], TestUtility.talks[0], TestUtility.talks[2]])
      })
    })
  })
})
