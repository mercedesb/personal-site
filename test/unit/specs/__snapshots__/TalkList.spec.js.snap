// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TalkList Snapshots matches snapshot 1`] = `
<div
  class="TalkList"
>
  <div
    class="TalkListItem TalkListItem--gray Draw--300"
    color="gray"
    givenat="[object Object]"
    icon=<svg></svg>
    iconcssclass="Draw--300"
    id="talk2"
    maincontent="talk main content"
    preamble="talk 2 preamble"
    title="talk 2 title"
    urlsegment="talk2"
  >
    <!---->
     
    <!---->
  </div>
  <div
    class="TalkListItem TalkListItem--gray Draw--300"
    color="gray"
    givenat="[object Object]"
    icon=<svg></svg>
    iconcssclass="Draw--300"
    id="talk1"
    maincontent="talk main content"
    preamble="talk 1 preamble"
    title="talk 1 title"
    urlsegment="talk1"
  >
    <!---->
     
    <!---->
  </div>
  <div
    class="TalkListItem TalkListItem--gray Draw--300"
    color="gray"
    givenat="[object Object]"
    icon=<svg></svg>
    iconcssclass="Draw--300"
    id="talk3"
    maincontent="talk main content"
    preamble="talk 3 preamble"
    title="talk 3 title"
  >
    <!---->
     
    <!---->
  </div>
</div>
`;
